File: ukidelly/PGS/240604/배열_만들기_3.java
import java.util.ArrayList;
import java.util.stream.IntStream;

// IntStream을 써봤는데 뭐랄까... 그냥 for문 쓰는게 더 편한 것 같네요
class Solution {
    public int[] solution(int[] arr, int[][] intervals) {

    ArrayList<Integer> result = new ArrayList<Integer>();

    for (int[] interval : intervals) {
      int left = interval[0];
      int right = interval[1];

      IntStream.rangeClosed(left, right).forEach(i -> result.add(arr[i]));
    }

    return result.stream().mapToInt(i -> i).toArray();
  }
}

File: ukidelly/PGS/240604/배열_만들기_3.kt
/**
 * 제 풀이
 */
class Solution {
  fun solution(arr: IntArray, intervals: Array<IntArray>): IntArray {
    var answer = intArrayOf()

    intervals.forEach { (start, end) -> answer += arr.slice(start..end) }

    return answer
  }
}

/**
 * 인상 깊었던 풀이
 */
class Solution {
    fun solution(arr: IntArray, intervals: Array<IntArray>): IntArray {
        return intervals.flatMap { (left, right) -> arr.slice(left..right) }.toIntArray()
    }
}